File: c#/1st_ex/main.cs
using System;

class Program
{
    static void Err(string exc)
    {
        throw new Exception(exc);
    }

    static int Init()
    {
        int val = int.Parse(Console.ReadLine());
        if (val <= 0)
        {
            Err("Invalid input");
        }
        return val;
    }

    static int Func(int S, int N, int M)
    {
        if (S * M > (S - S / 7) * N)
        {
            return -1;
        }
        return (int)Math.Ceiling((double)S * M / N);
    }

    static void Main(string[] args)
    {
        int S = Init();
        int N = Init();
        int M = Init();

        Console.WriteLine(Func(S, N, M));
    }
}

File: ruby/1st_ex/main.rb
def err(exc)
  raise Exception.new(exc)
end

def init
  val = gets.to_i
  if val <= 0
    err("Invalid input")
  end
  val
end

def func(s, n, m)
  if s * m > (s - s / 7) * n
    return -1
  end
  return (s * m / n.to_f).ceil
end

s = init()
n = init()
m = init()

puts func(s, n, m)
